common/context.go

```diff
@@ -2,15 +2,21 @@ package common
 
 import (
 	"context"
+	"fmt"
+	"math/rand"
+	"time"
 )
 
 type ctxKey int
 
 const (
 	ctxKeyLaunchOptions ctxKey = iota
 	ctxKeyHooks
+	ctxKeyTraceID
 )
 
+var rnd = rand.New(rand.NewSource(time.Now().UnixNano())) //nolint
```

Comment:
I'm not happy with this global, but I don't think we need to pass it somewhere from `BrowserType`, and it would be slightly awkward, so I don't mind it much.

Though I'm currently running into an issue on this branch after a few minutes of stress testing that ends with:

```
panic: GoError: launching browser: inotify_init() failed: Too many open files (24)
```

... which is strange, since a) my `ulimit -n` is set to `50000`, and b) I don't see how the changes here could cause it, unless this line does it somehow. Initially I had this inside `WithTraceID()`, and thought that might be the issue, but I moved it to a global, and still see the error. I haven't run into it on `main`, so I'm skeptical it's something here.

Any thoughts?

Reply:
OK, I moved it to `NewBrowserType()` in f01e97e.

Still, please (stress) test this and let me know if you run into any issues.